### ISSUE TITLE
lzo.lib file not found error in Windows 10

### ISSUE THREAD

**Issue by jerokpradeep** [jd-boyd/python-lzo#49]
Hi,
While trying to install this I'm getting this error. The lzo.lib is not found anywhere in the library as well under the src folder. Please help as I know I'm doing something worn over here.

LINK : fatal error LNK1181: cannot open input file 'C:\src\lzo-2.09\lzo.lib'
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\link.exe' failed with exit status 1181

Thanks
Pradeep K

**Comment by Tuo-ZHANG** [comment]
Same issue here, have you solved it?

**Comment by maxgerhardt** [comment]
I've gotten it it to work with the following procedure (prerequisites: VisualStudio + CMake install): 

1. Download and extract the .tar.gz file as noted in https://github.com/jd-boyd/python-lzo#windows
2. Extract it in `C:\src\lzo-2.09\*source files here*`
3. Open a new shell in the directory above
4. `mkdir build && cd build`
5. `cmake ..`
6. Project files for VisualStudio should be generated. Open `lzo.sln`. 
7. Build the `lzo_static` sub-project
8. You should now have a `C:\src\lzo-2.09\build\Release\lzo2.lib`
9. Duplicate above file as `C:\src\lzo-2.09\lzo.lib` 
10. Open a new shell where you've downloaded Python-LZO
11. `set LZO_DIR=C:\src\lzo-2.09`
12. `python.exe setup.py install`
13. Should now end with `Finished processing dependencies for ubidump==1.0.0` without error.

```
>python ubidump.py
usage: ubidump.py [-h] [--savedir DIRECTORY] [--preserve] [--cat FILE] [--listfiles] [--dumptree] [--verbose]
                  [--debug] [--encoding ENCODING] [--masteroffset MASTEROFFSET] [--root ROOT] [--rawdump]
                  [--volume VOLUME] [--hexdump LEB:OFF:N] [--nodedump LEB:OFF]
                  FILES [FILES ...]
ubidump.py: error: the following arguments are required: FILES
```

**Comment by Ozzy4201** [comment]
> 

How can i hopen a shell where i have downloaded Python-LZO to set the LZO_DIR when i need to set the LZO_DIR to install python-lzo?

**Comment by maxgerhardt** [comment]
You can download Python-LZO without installing it, I just meant you `git clone` **this** repository (or [zip downloaded it](https://github.com/jd-boyd/python-lzo/archive/refs/heads/master.zip)).

**Comment by jd-boyd** [comment]
This should be fixed with 1.15.

**Comment by endless-creation** [comment]
I encountered this problem trying to install python-lzo==1.15 with lzo version 2.10

**Comment by maxgerhardt** [comment]
Encountered meaning you encountered **and** solved it? Installing python-lzo 1.15 still works fine for me. Camek and Visual Studio 2022 Build Tools (with C++ Desktop tools installed), I used this method in a Developer Terminal for VS 2022, with a base path of `C:\Users\Max\temp\`.

![grafik](https://github.com/user-attachments/assets/cd98383f-cded-4e3e-936e-8c27cff5f15d)

(Adapt paths as needed.)

```
git clone -b "v1.15" https://github.com/jd-boyd/python-lzo.git
cd python-lzo
cd lzo-2.10
cmake -G "Visual Studio 17 2022" ..
msbuild lzo_static_lib.vcxproj -p:Configuration=Release;Platform=x64;OutDir=C:\Users\Max\temp\python-lzo\lzo-2.10\
```
That should have generated `C:\Users\Max\temp\python-lzo\lzo-2.10\lzo2.lib`

Then just
```
pip install .
```
in the original python-lzo directory (`C:\Users\Max\temp\python-lzo\`) in this example.

**Comment by endless-creation** [comment]
Yes correct, I basically followed those steps with a couple differences -- I just wasn't aware that building the lzo2.lib was required, it wasn't clear from the README. Thanks

**Comment by sea-ice** [comment]
> Encountered meaning you encountered **and** solved it? Installing python-lzo 1.15 still works fine for me. Cmake and Visual Studio 2022 Build Tools (with C++ Desktop tools installed), I used this method in a Developer Terminal for VS 2022, with a base path of `C:\Users\Max\temp\`.
> 
> ![grafik](https://private-user-images.githubusercontent.com/26485477/373886157-cd98383f-cded-4e3e-936e-8c27cff5f15d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwNTE5MjIsIm5iZiI6MTczNjA1MTYyMiwicGF0aCI6Ii8yNjQ4NTQ3Ny8zNzM4ODYxNTctY2Q5ODM4M2YtY2RlZC00ZTNlLTkzNmUtOGMyN2NmZjVmMTVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDA0MzM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQwNTM2NTU2OWMxMGEyN2I3Y2YxZjBmOWY2OTI0NTQzZmFmZjc1YzA3ODM1ZmQ5OGYwYTQ0Zjk4NGM4NDlkMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bzwyca_9bJxOfOOF5E3qh5YhfKhfdmq2sJjyWHhojKk)
> 
> (Adapt paths as needed.)
> 
> ```
> git clone -b "v1.15" https://github.com/jd-boyd/python-lzo.git
> cd python-lzo
> cd lzo-2.10
> mkdir build && cd build
> cmake -G "Visual Studio 17 2022" ..
> msbuild lzo_static_lib.vcxproj -p:Configuration=Release;Platform=x64;OutDir=C:\Users\Max\temp\python-lzo\lzo-2.10\
> ```
> 
> That should have generated `C:\Users\Max\temp\python-lzo\lzo-2.10\lzo2.lib`
> 
> Then just
> 
> ```
> pip install .
> ```
> 
> in the original python-lzo directory (`C:\Users\Max\temp\python-lzo\`) in this example.

I followed those steps on windows 11 but linking failed when I pip install:

E:\python-lzo>pip install .
Defaulting to user installation because normal site-packages is not writeable
Processing e:\python-lzo
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: python-lzo
  Building wheel for python-lzo (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for python-lzo (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [59 lines of output]
      C:\Users\Joker\AppData\Local\Temp\pip-build-env-9y7d19_a\overlay\Lib\site-packages\setuptools\_distutils\dist.py:261: UserWarning: Unknown distribution option: 'tests_require'
        warnings.warn(msg)
      LZO_DIR: .\lzo-2.10
      lzomodule.c
      lzomodule.c(610): warning C4141: 'dllexport': used more than once
         Creating library build\temp.win32-cpython-313\Release\lzo.cp313-win_amd64.lib and object build\temp.win32-cpython-313\Release\lzo.cp313-win_amd64.exp
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyBytes_AsString
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1y_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyEval_SaveThread
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1x_optimize
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1f_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyLong_FromLong
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo_version_date
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyBytes_FromString
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1z_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo2a_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1f_1_compress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyDict_SetItemString
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1a_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1x_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyErr_SetString
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1y_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol ___lzo_init_v2
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyErr_Format
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo_crc32
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo_adler32
      lzomodule.obj : error LNK2001: unresolved external symbol __imp___Py_Dealloc
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyModule_GetDict
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1b_1_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1_99_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1_decompress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1y_1_compress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp___PyBytes_Resize
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1a_99_compress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyModule_Create2
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyErr_NewException
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1x_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1x_1_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1b_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1c_999_compress
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1c_1_compress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyMem_Free
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyErr_NoMemory
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1z_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyBytes_FromStringAndSize
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyArg_ParseTupleAndKeywords
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1a_decompress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyMem_Malloc
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1f_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo2a_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo_version_string
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1b_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo_version
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1c_decompress_safe
      lzomodule.obj : error LNK2001: unresolved external symbol _lzo1_compress
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyArg_ParseTuple
      lzomodule.obj : error LNK2001: unresolved external symbol __imp__PyEval_RestoreThread
      build\lib.win32-cpython-313\lzo.cp313-win_amd64.pyd : fatal error LNK1120: 51 unresolved externals
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.42.34433\\bin\\HostX86\\x86\\link.exe' failed with exit code 1120
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for python-lzo
Failed to build python-lzo
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (python-lzo)
